File: trees/ltcd_199.cpp
//LeetCode - problem 199

    void dfs(TreeNode* root, int level, vector<int>&res){
        if(!root)
            return;
        if(res.size()==level) res.push_back(root->val);
        dfs(root->right,level+1,res);
        dfs(root->left,level+1,res);
    }

    vector<int> rightSideView(TreeNode* root){
        vector<int> res;
        dfs(root,0,res);
        return res;
    }

File: trees/ltcd_199.py
'''
LeetCode - problem 199
'''

def rightSideView(self, root):
    ans = []
    self.h = -1
    def prog(root,height):
        if root is None:
            return
        if height > self.h:
            ans.append(root.val)
            self.h = height
        prog(root.right, height + 1)
        prog(root.left, height + 1)
    prog(root,0)
    return ans
